crates/orchestrator/src/network/node.rs

```diff
@@ -273,6 +273,22 @@ impl NetworkNode {
         Ok(())
     }
 
+    /// Run a script inside the node's container/environment
+    ///
+    /// The script will be uploaded to the node, made executable, and executed with
+    /// the provided arguments and environment variables.
+    ///
+    /// Returns `Ok(stdout)` on success, or `Err((exit_status, stderr))` on failure.
+    pub async fn run_script(
+        &self,
+        options: provider::types::RunScriptOptions,
```

Comment:
Let's bring those to the scope with
```rs
use provider::types::{RunScriptOptions, ExecutionResult};
```

```suggestion
        options: RunScriptOptions,
```